api/src/main/java/com/netflix/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -94,6 +94,10 @@ public R or(R leftResult, R rightResult) {
       return null;
     }
 
+    public R startsWith(BoundReference<String> ref, Literal<String> lit) {
+      return null;
```

Comment:
I think this should throw an `UnsupportedOperationException`. The problem is that the all of the visitors were written without this method and we need to ensure that either they support `startsWith` or fail when it is encountered.

In fact, the other methods probably should too, but that doesn't need to be done in this PR.

Reply:
This is something I didn't know/understand. Now I see why it returns `nulls`. 
My question now is how to choose which visitors should implement this predicate? For example, should the {Inclusive|Strict}MetricsEvaluator, that checks if a DataFile contains some rows or not, rely on actually checking rows? or just metadata, e.g., checking the dictionary of encoded values? or both (metadata and data)?
Anyway, I will leave as it is for now then, and open a different issue for that.

Reply:
This PR definitely needs to throw an exception, or else it isn't clear what the behavior of those implementations will be when a node returns null. If an implementation doesn't support a predicate and it is passed that predicate, it should fail.

Reply:
I updated it as well.